panic oob

```
position fen qqqqkqqq/8/8/8/8/8/8/QQQQKQQQ w - - 0 1
go depth 30
info depth 0 score cp 951 nodes 12016 time 10 hashfull 0 pv
info depth 1 score cp 951 nodes 22989 time 15 hashfull 0 pv c1c8
info depth 2 score cp 951 nodes 34699 time 20 hashfull 0 pv c1c8 b8c8
info depth 3 score cp 951 nodes 49611 time 25 hashfull 0 pv c1c8 b8c8 d1d8
info depth 4 score cp 543 nodes 66844 time 33 hashfull 0 pv c1c8 b8c8 a1a8 c8a8
info depth 5 score cp 986 nodes 81537 time 40 hashfull 0 pv f1f8 g8f8 a1h8 a8h1 d1d8
info depth 6 score cp 1091 nodes 107172 time 53 hashfull 2 pv f1f8 g8f8 c1e3 h8e5 b1g6 f8f7
info depth 7 score cp 1091 nodes 135819 time 66 hashfull 3 pv f1f8 g8f8 c1e3 h8e5 b1g6 f8f7 a1e5
info depth 8 score cp 1086 nodes 202470 time 108 hashfull 10 pv c1e3 g8e6 b1g6 e8e7 f1f8 h8f8 h1a8
panic: runtime error: index out of range [101] with length 101

goroutine 16 [running]:
github.com/paulsonkoly/chess-3/search.lmr(...)
        /Users/phaul/chess-3/search/search.go:447
github.com/paulsonkoly/chess-3/search.AlphaBeta(0x14000104020, 0x659, 0x75e, 0x3, 0x6, 0x1, 0x0, 0x1400007e0a0)
        /Users/phaul/chess-3/search/search.go:331 +0x2118
github.com/paulsonkoly/chess-3/search.AlphaBeta(0x14000104020, 0xf8a2, 0xf9a7, 0x4, 0x5, 0x1, 0x0, 0x1400007e0a0)
        /Users/phaul/chess-3/search/search.go:347 +0x1574
github.com/paulsonkoly/chess-3/search.AlphaBeta(0x14000104020, 0x659, 0x75e, 0x5, 0x4, 0x1, 0x0, 0x1400007e0a0)
        /Users/phaul/chess-3/search/search.go:347 +0x1574
github.com/paulsonkoly/chess-3/search.AlphaBeta(0x14000104020, 0xf8a2, 0xf9a7, 0x6, 0x3, 0x1, 0x0, 0x1400007e0a0)
        /Users/phaul/chess-3/search/search.go:347 +0x1574
github.com/paulsonkoly/chess-3/search.AlphaBeta(0x14000104020, 0x659, 0x75e, 0x7, 0x2, 0x1, 0x0, 0x1400007e0a0)
        /Users/phaul/chess-3/search/search.go:347 +0x1574
github.com/paulsonkoly/chess-3/search.AlphaBeta(0x14000104020, 0xf8a2, 0xf9a7, 0x8, 0x1, 0x1, 0x0, 0x1400007e0a0)
        /Users/phaul/chess-3/search/search.go:347 +0x1574
github.com/paulsonkoly/chess-3/search.AlphaBeta(0x14000104020, 0x40c, 0x75e, 0x9, 0x0, 0x1, 0x0, 0x1400007e0a0)
        /Users/phaul/chess-3/search/search.go:347 +0x1574
github.com/paulsonkoly/chess-3/search.Search(0x14000104020, 0x1e, 0x1400007e0a0)
        /Users/phaul/chess-3/search/search.go:105 +0x504
github.com/paulsonkoly/chess-3/uci.(*Engine).Search(...)
        /Users/phaul/chess-3/uci/uci.go:361
github.com/paulsonkoly/chess-3/uci.(*Engine).handleGo.func1()
        /Users/phaul/chess-3/uci/uci.go:316 +0x3c
created by github.com/paulsonkoly/chess-3/uci.(*Engine).handleGo in goroutine 15
        /Users/phaul/chess-3/uci/uci.go:315 +0x4c4
```

--- a/search/search.go
+++ b/search/search.go
@@ -444,7 +444,7 @@ var log = [...]int{
 // x = (1..200).map {|i| (Math.log2(i) * 69).round }.unshift(0)
 // 10.times.map {|d| 30.times.map {|m| (x[d] * x[m] )>>14}}
 func lmr(d Depth, mCount int, improving, pvN, cutN bool) Depth {
-	value := (log[d] * log[mCount]) >> 14
+	value := (log[d] * log[min(mCount, len(log) - 1)]) >> 14
 
 	// if !quiet {
 	// 	value /= 2

--- a/search/search_test.go
+++ b/search/search_test.go
@@ -12,6 +12,30 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
+func TestAlphaBeta(t *testing.T) {
+	tests := []struct {
+		name string
+		b    *board.Board
+		d    Depth
+	}{
+		{name: "position fen qqqqkqqq/8/8/8/8/8/8/QQQQKQQQ w - - 0 1",
+			b: Must(board.FromFEN("qqqqkqqq/8/8/8/8/8/8/QQQQKQQQ w - - 0 1")),
+			d: 9,
+		},
+	}
+	sst := search.NewState(1)
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			sst.Clear()
+			assert.NotPanics(t,
+				func() {
+					search.Search(tt.b, tt.d, sst)
+				},
+			)
+		})
+	}
+}
+
 func TestQuiescence(t *testing.T) {
 	tests := []struct {
 		name string // description of this test case

